src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/Connection/PreloginSubHandlers/BaseTlsHandler.cs

```diff
@@ -0,0 +1,139 @@
+                    new ServerInfo(request.ConnectionContext.ConnectionString));
+                // TODO; enhance
+                throw request.Exception;
+            }
+
+            LogWarningIfNeeded(request);
+
+            static void LogWarningIfNeeded(PreLoginHandlerContext request)
+            {
+                string warningMessage = request.ConnectionContext.SslStream.SslProtocol.GetProtocolWarning();
+                if (!string.IsNullOrEmpty(warningMessage))
+                {
+                    if (ShouldNotLogWarning(request))
+                    {
+                        // Skip console warning
+                        SqlClientEventSource.Log.TryTraceEvent("<sc|{0}|{1}|{2}>{3}",
+                            nameof(PreloginHandler),
+                            nameof(AuthenticateClientInternal),
+                            SqlClientLogger.LogLevel.Warning,
+                            warningMessage);
+                    }
+                    else
+                    {
+                        // This logs console warning of insecure protocol in use.
+                        request.ConnectionContext.Logger.LogWarning(nameof(PreloginHandler), nameof(AuthenticateClientInternal), warningMessage);
+                    }
+                }
+            }
+
+            static bool ShouldNotLogWarning(PreLoginHandlerContext request)
+            {
+                return !request.ConnectionEncryptionOption && LocalAppContextSwitches.SuppressInsecureTLSWarning;
+            }
+        }
+
+        private async ValueTask AuthenticateClient(PreLoginHandlerContext context, bool isAsync, CancellationToken ct)
+        {
+            try
+            {
+                SqlClientEventSource.Log.TryTraceEvent("PreloginHandler.AuthenticateClient | Info | Session Id {0}", context.ConnectionContext.ConnectionId);
+                using (TrySNIEventScope.Create(nameof(PreloginHandler)))
+                {
+                    Guid _connectionId = context.ConnectionContext.ConnectionId;
+                    ValidateCert = context.ValidateCertificate;
+                    string serverName = context.ConnectionContext.DataSource.ServerName;
+                    SslOverTdsStream sslOverTdsStream = context.ConnectionContext.SslOverTdsStream;
+                    SslStream sslStream = context.ConnectionContext.SslStream;
+                    try
+                    {
+                        SslClientAuthenticationOptions options =
+                            context.IsTlsFirst ?
+                                new()
+                                {
+                                    TargetHost = serverName,
+                                    ClientCertificates = null,
+                                    EnabledSslProtocols = s_supportedProtocols,
+                                    CertificateRevocationCheckMode = X509RevocationMode.NoCheck,
```

Comment:
Although Setting to NoCheck can solve some issues for MAUI or other phone applications, but does not sound like a good practice. This will be flagged by CodeQL first thing first. Is it possible to make it optional for the users by something like an AppContext switch or some other mechanism?

Reply:
We have has requests for ability to provide revocation list before, so would recommend opening a feature request for tracking this in future, and adding a note in code for now.

Reply:
The certificate chain can have certificates which have a link to revoked certificate.
I know that TLS cert handling on Windows definitely checks for the revocation list. 
Let me experiment with the APIs on .Net to see if I can get this to honor the Revocation list in the cert chain. We wont need a feature, but it is possible, that honoring the CRL is a breaking change. Perhaps an App Context Switch ? 

@JRahnama  Thanks for pointing this out.

Reply:
Added a TODO. More investigation needed.